machine/machine_core/machine_virtual.py

```diff
@@ -203,7 +214,8 @@ def interface(self, number=None):
         }
         return result
 
-    def host(self, number=None, restrict=False, isolate=False, forward={}):
+    def host(self, number: Optional[int] = None, restrict: bool = False, isolate: bool = False,
```

Comment:
If it won't fit on one line, then please do one of these:

 - PEP8 hanging indent style like this:

```python
def host(
    self, number, restrict, ...
) -> None:
```

 - PEP8 hanging indent style like this:

```python
def host(
    self,
    number,
    restrict,
    ...
) -> None:
```

 - visual indent style like this:

```python
def host(self,
         number,
         restrict,
         ...) -> None:
```